exponentation_by_squaring/exponentation_by_squaring.c

```diff
@@ -0,0 +1,25 @@
+#include<stdio.h>
+
+/* Algorithm to calculate x raised to power k very efficiently
+*  Can be used as a great alternative to normal pow function
+*  base - base of the expression
+*  power - power of expression
+*/
+
+long int exponentation_by_squaring(int base, int power) {
```

Comment:
I guess maybe its some problem with my system.
It seems to work fine on online compiler's too.